invenio/modules/records/access.py

```diff
@@ -133,7 +163,7 @@ def check_user_can_view_record(user_info, recid):
         return (0, '')
 
     restricted_collections = get_restricted_collections_for_recid(
-        recid, recreate_cache_if_needed=False
+        recid, recreate_cache_if_needed=True
```

Comment:
this might slow down things badly. do you recall the reason why did you change it?